src/styles/components/_buttons.scss

```diff
@@ -138,3 +108,61 @@
 .w-button--secondary:active::after {
   background-color: rgba($BLACK, .1);
 }
+
+// A button with text and an icon.
+.w-button--with-icon {
+  padding: 0 24px 0 16px;
+}
+
+// A button with only an icon.
+.w-button--icon {
+  @include base-button();
+  color: $GREY_700;
+  height: 3rem;
+  overflow: unset;
+  padding: 0;
+  transition: background $TRANSITION_SPEED ease, color $TRANSITION_SPEED ease;
+  width: 3rem;
+}
+
+@include hover() {
+  .w-button--icon:hover,
+  .w-button--icon:focus {
```

Comment:
It might be worth commenting that we leave the default `outline: ...` alone (the one implicitly defined on `:focus`).

Reply:
Actually, we don't leave it alone. `.w-button` doesn't currently have an accessible focus style. It just shows a slightly darker version of the hover overlay that isn't dark enough to meet WCAG 2.1 minimum contrast requirements for UI (3:1).

I've fixed that for circular buttons by adding a 1px `box-shadow`, which I'm happy to note. Also happy to suggest an accessible focus style for `.w-button` in a separate PR if folks are interested. 😄

Reply:
Something like these, maybe:

![image](https://user-images.githubusercontent.com/14199217/75723599-79b2dd80-5c91-11ea-8900-d85d88ae3c96.png)

![image](https://user-images.githubusercontent.com/14199217/75723586-74ee2980-5c91-11ea-8b70-769ec766081c.png)

(I used a psuedo with the opacity dropped a bit so the outlines aren't so slap-you-in-the-face while still being accessible. I'm thinking this would only apply on keyboard focus, as with our other focus rings. I'd also like to turn off our current button focus style for non-keyboard focus so it doesn't linger. I know it's inherited from Material, but the drop shadow and lighter background make it look to me like something's gone wrong with the button's state.)

Reply:
I'm not really sure what I'm looking at, sorry. Is the blue background the `:focus` hint?

Can `:focus` and `:hover` have the same colors, by the spec? I can't see where they're different.

Reply:
Sorry, the inset outline is the focus hint here. I'm showing both `.w-button` and `.w-button--primary` variants.

And, yeah, so far as I know there's no rule against `:hover` and `:focus` being the same, assuming your `:hover` state is contrast conformant. That's often not the case, though, since hover is a transient state. (In fact, our `:hover` isn't conformant.) So, one common strategy is to use an additive `:focus` style: `:hover` plus an additional hint. That's what I'm proposing here, with the inset outline being the additional hint.